Gateway/Http/Client/Payment.php

```diff
@@ -50,10 +48,19 @@ public function __construct(
      */
     public function placeRequest(TransferInterface $transferObject)
     {
+
         $this->omise->defineUserAgent();
         $this->omise->defineApiVersion();
         $this->omise->defineApiKeys();
+        
+        $transferObjectBody = $transferObject->getBody();
+
+        // if charge_id already exists than action is 'manual capture'
+        if ($chargeID = $transferObjectBody['charge_id']) {
+            $charge = $this->apiCharge->find($chargeID);
+            return ['charge' => $charge->capture()];
```

Comment:
This doesn't feel a right place and I feel the code smell..
`Omise\Payment\Gateway\Http\Client\Payment` is a class that shared between all payment methods (Alipay, Credit Card, Installment, Internet Banking, Tesco Bill Payment).
I think there might be a better place than having having a condition for a specific case like card-capture at this class (cannot tell at the moment, I will have a look a bit more).

Reply:
+1

Reply:
it is payment, payment might be: create `charge` or  payment `capture`.

we have following payment actions:
`authorize`, `authorize_capture` or `captured` previously authorized

Until last time this Class was responsible only for `authorize` or `authorize_capture`, because `capture` was not implemented yet. Now it is.
Yes it is shared by all methods, and it is fine to put it here, in the future I believe there will be more methods that will do `authorize` and than `capture` payment action.